regression-tests/run-tests.sh

```diff
@@ -0,0 +1,306 @@
+#!/bin/bash
+
+################
+usage() {
+    echo "Usage: $0 -c <compiler> [-t <tests to run>]"
+    echo "    -c <compiler>     The compiler to use for the test"
+    echo "    -t <tests to run> Runs only the provided, comma-separated tests (filenames including .cpp2)"
+    echo "                      If the argument is not used all tests are run"
+    exit 1
+}
+
+check_file () {
+    file="$1"
+    # Check if the file is tracked by bit
+    git ls-files --error-unmatch "$file" > /dev/null 2>&1
+    untracked=$?
+    # Compare the content with the refernece value
+    diff_output=$(git diff --ignore-cr-at-eol -- "$file")
+}
+
+optstring="c:t:"
+while getopts ${optstring} arg; do
+  case "${arg}" in
+    c)
+        cxx_compiler="${OPTARG}"
+        ;;
+    t)
+        # Replace commas with spaces
+        chosen_tests=${OPTARG/,/ }
+        ;;
+    \?)
+        echo "Invalid option: -${OPTARG}."
+        echo
+        usage
+        ;;
+    :)
+        echo "Missing option argument for -$OPTARG"
+        echo
+        usage
+        ;;
+    *)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [ -z "$cxx_compiler" ]; then
+    echo "Compiler not specified"
+    usage
+fi
+
+tests=$(ls | grep ".cpp2$")
+if [[ -n "$chosen_tests" ]]; then
+    for test in $chosen_tests; do
+        if ! [[ -f "$test" ]]; then
+            echo "Requested test ($test) not found"
+            exit 1
+        fi
+    done
+    echo "Performing tests:"
+    for test in $chosen_tests; do
+        echo "    $test"
+    done
+    echo
+    tests="$chosen_tests"
+else
+    printf "Performing all regression tests\n\n"
+fi
+
+expected_results_dir="test-results"
+
+################
+# Get the directory with the exec outputs and compilation command
+if [[ "$cxx_compiler" == *"cl.exe"* ]]; then
+    compiler_cmd='cl.exe -nologo -std:c++latest -MD -EHsc -I ..\include -I ..\..\..\include -experimental:module -Fe:'
```

Comment:
The 2 include paths are necessary due to the fact that compilation has to be requested in two different folders due to the current setup. This could be improved, but will require small modification in the tests result files.
With that modification, if compilation of the generated C++ is performed in the `regression-tests` directory, the error outputs from cppfront are more detailed and include pointers into the cpp2 source. This is because the cpp2 files are found in that directory.
I didn't want to modify a lot of files in this PR. I can create another PR that will explain better what I mean here.

Reply:
Why don't you use an absolute include path?

Reply:
I tried. The thing is the path may show up in test results, on compilation errors.